web/src/stream_card_popover.ts

```diff
@@ -0,0 +1,77 @@
+import $ from "jquery";
+import assert from "minimalistic-assert";
+import type * as tippy from "tippy.js";
+
+import render_stream_info_popover from "../templates/popovers/stream_info_popover.hbs";
+
+import * as browser_history from "./browser_history";
+import * as hash_util from "./hash_util";
+import * as popover_menus from "./popover_menus";
+import {current_user} from "./state_data";
+import * as stream_data from "./stream_data";
+import * as sub_store from "./sub_store";
+import * as ui_util from "./ui_util";
+
+let stream_info_popover_instance: tippy.Instance | undefined;
+
+function is_open_stream_info(): boolean {
+    return Boolean(stream_info_popover_instance);
+}
+
+function hide_stream_info_popover(): void {
+    if (stream_info_popover_instance !== undefined) {
+        stream_info_popover_instance.destroy();
+        stream_info_popover_instance = undefined;
+    }
+}
+
+export function toggle_stream_card_popover(element: tippy.ReferenceElement): void {
+    if (is_open_stream_info()) {
+        hide_stream_info_popover();
+        return;
+    }
```

Comment:
We can simplify all this logic by using `popover_menus.register_popover_menu()` instead, take a look at `topic_popover.js` for context. We eventually want to move all the popovers to use this logic for simplification, but since this a new addition, we might as well start with it.